packages/astro/src/i18n/index.ts

```diff
@@ -386,6 +386,10 @@ export function redirectToFallback({
 				if (pathFallbackLocale === defaultLocale && strategy === 'pathname-prefix-other-locales') {
 					if (context.url.pathname.includes(`${base}`)) {
 						newPathname = context.url.pathname.replace(`/${urlLocale}`, ``);
+						// Ensure the pathname are non-empty. Redirects like "/fr" => "" may create infinite loops.
```

Comment:
Can you please add a more generic comment? For example, the `/fr` part has no context and it isn't related to the fix